hypervisor/boot/sbl/abl_seed_parse.c

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (C) 2018 Intel Corporation. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <hypervisor.h>
+#include <abl_seed_parse.h>
+
+#define ABL_SEED_LEN 32U
+struct abl_seed_info {
+	uint8_t svn;
+	uint8_t reserved[3];
+	uint8_t seed[ABL_SEED_LEN];
+};
+
+#define ABL_SEED_LIST_MAX 4U
+struct dev_sec_info {
+	uint32_t size_of_this_struct;
+	uint32_t version;
+	uint32_t num_seeds;
+	struct abl_seed_info seed_list[ABL_SEED_LIST_MAX];
+};
+
+static const char *boot_params_arg = "dev_sec_info.param_addr=";
+
+static void parse_seed_list_abl(void *param_addr)
+{
+	uint32_t i;
+	uint32_t legacy_seed_index = 0;
+	struct seed_info dseed_list[BOOTLOADER_SEED_MAX_ENTRIES];
+	struct dev_sec_info *sec_info = (struct dev_sec_info *)param_addr;
+
+	if (sec_info == NULL)
+		goto fail;
+
+	/*
+	 * The seed_list from ABL contains several seeds which based on SVN
+	 * and one legacy seed which is not based on SVN. The legacy seed's
+	 * svn is the minimum in the seed list. And CSE ensures at least two
+	 * seeds will be generated which will contain the legacy seed.
+	 * Here find the legacy seed index first.
+	 */
+	for (i = 0U; i < sec_info->num_seeds; i++) {
```

Comment:
use 1 instead of 0 here

Reply:
why 1?

Reply:
legacy_seed_index initial value is 0, so i starts from 1 is correct.